utilities/generate_messages_catalog.py

```diff
@@ -131,96 +134,172 @@ def _is_callable(item):
                             ]
-                    msgArg = item.args[1 + args_offset]
-                    if isinstance(msgArg, ast.Str):
-                        msg = msgArg.s
-                    elif ((isinstance(msgArg, ast.Call) and
-                           getattr(msgArg.func, "id", '') == '_')):
-                        msg = msgArg.args[0].s
-                    elif ((any(isinstance(element, (ast.Call,ast.Name))
-                           for element in ast.walk(msgArg)))):
-                        msg = "(dynamic)"
                     else:
-                        continue # not sure what to report
-                    keywords = []
-                    for keyword in item.keywords:
-                        if keyword.arg == 'modelObject':
-                            pass
-                        elif keyword.arg == 'messageCodes':
-                            msgCodeArg = keyword.value
-                            if ((any(isinstance(element, (ast.Call, ast.Name))
-                                 for element in ast.walk(msgCodeArg)))):
-                                pass  # dynamic
-                            else:
-                                msgCodes = [
-                                    element.s
-                                    for element in ast.walk(msgCodeArg)
-                                    if isinstance(element, ast.Str)
-                                ]
-                        else:
-                            keywords.append(keyword.arg)
-                    for msgCode in msgCodes:
-                        id_messages.append(
-                            {
-                                'message_code': msgCode,
-                                'message': entityEncode(msg),
-                                'level': level,
-                                'keyword_arguments': entityEncode(
-                                    " ".join(keywords)
-                                ),
-                                'reference_filename': refFilename,
-                                'line_number': item.lineno
-                            }
-                        )
-                except (AttributeError, IndexError):
-                    pass
-    return id_messages, file_count
+                        keywords.append(keyword.arg)
+                for msgCode in msgCodes:
+                    id_messages.append(
+                        {
+                            'message_code': msgCode,
+                            'message': entity_encode(msg),
+                            'level': level,
+                            'keyword_arguments': entity_encode(
+                                " ".join(keywords)
+                            ),
+                            'reference_filename': ref_module_name,
+                            'line_number': item.lineno
+                        }
+                    )
+            except (AttributeError, IndexError):
```

Comment:
With all the places we are checking for attributes and indices, why do we need to except on attribute and index failures?